File: Leetcode/1161. Maximum Level Sum of a Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        queue<TreeNode*> q;
        q.push(root);
        int res=1;
        int level=1;
        int sum=root->val;
        while(!q.empty()){
            int size=q.size();
            int temp=0;
            while(size--){
                TreeNode* t=q.front();
                temp+=t->val;
                if(t->right!=nullptr){
                    q.push(t->right);
                }
                if(t->left!=nullptr){
                    q.push(t->left);
                }
                q.pop();
            }
            if(sum<temp){
                sum=temp;
                res=level;
            }
            level++;
        }
        return res;
    }
};

File: Leetcode/1161. Maximum Level Sum of a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        q=deque()
        q.append(root)
        res=1
        mx=root.val
        l=1
        while q:
            s=len(q)
            a=0
            while s>0:
                t=q[0]
                q.popleft()
                if t.right:
                    q.append(t.right)
                if t.left:
                    q.append(t.left)
                a+=t.val
                s-=1
            if a>mx:
                mx=a
                res=l
            l+=1
        return res
